### DISCUSSION TITLE
Feature Request: Upstreaming creation of struct and enum types using C API

### DISCUSSION THREAD

**Discussion by alnkesq** [duckdb/duckdb#8668] [Ideas]
I recently published a .NET interoperability layer for DuckDB ([DuckDbSharp](https://github.com/alnkesq/DuckDbSharp)), which unlike existing ones, supports structs and sublists. However, it requires two new small functions in the C API.

I already have a [small patch](https://github.com/alnkesq/DuckDbSharp/blob/main/src/duckdb.patch) that wraps the C++ type creation functions, and I was wondering if something like that could be upstreamed.
For one of those two functions, I opened a [PR](https://github.com/duckdb/duckdb/pull/8455/) some time ago, but I encountered (apparently unrelated) issues with CI and I'm not very familiar with this codebase, so I opened a feature request instead.

**Comment by Mause** [comment]
I believe this is now completed
